chevah/keycert/ssh.py

```diff
@@ -690,6 +690,7 @@ def _fromString_PUBLIC_OPENSSH(cls, data):
         @raises BadKeyError: if the blob type is unknown.
         """
         blob = base64.decodestring(data.split()[1])
+        assert cls._guessStringType(blob) == 'blob'
```

Comment:
just raise BadKeyError('Helpful unique message') instead of this generic assert.

we don't use assertion in this way as I consider it bad practice.

but I don't know if this is the right way to fix it.

_guessStringType is just a soft check... you might have a data of type blob but with other type of encoding errors.

We should update _fromString_BLOB and have it raise BadKeyError on errors

your implementation breaks the docstring as it will not raise AssertionError instead of BadKeyError